palette/src/luma/luma.rs

```diff
@@ -675,6 +675,13 @@ where
     }
 }
 
+impl<S, T> Eq for Luma<S, T>
+where
+    T: Component + Eq,
+    S: LumaStandard + Eq,
```

Comment:
We don't really need to require this. It's already guaranteed to be the same type, and no comparison has to happen there. The same for all the other spaces, really, so the places where it may have been derived could also be changed.